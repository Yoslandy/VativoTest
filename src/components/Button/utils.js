export const sizes = {
    small: "h-8 w-[100px]",
    big: "h-12 w-full",
}

export const types = {
    primary:
        "bg-primary-60 text-white active:bg-primary-100 disabled:bg-neutral-gray-70",
    secondary:
        "bg-transparent bg-primary-60 border-2 border-primary-60 text-primary-60 active:border-primary-100 active:text-primary-100 hover:border-primary-100 hover:text-primary-100 disabled:transparent disabled:text-neutral-gray-70 disabled:border-neutral-gray-70 disabled:shadow-none",
    warning: "bg-red-500 text-white active:bg-primary-100",
    success: "bg-emerald-500 text-white active:bg-primary-100",
    outline:
        "bg-white text-primary-60 border-2 border-primary-60 active:bg-primary-100",
    enabled: "bg-primary-100 text-white",
    tertiary: "bg-primary-100 text-white",
}
