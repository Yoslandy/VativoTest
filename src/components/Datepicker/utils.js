export const days = Array.from({ length: 31 }, (v, k) => k + 1).map(d => ({
    label: d.toString(),
    value: d,
}))

export const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
].map((m, i) => ({
    label: m,
    value: i + 1,
}))

const arrayOfYears = () => {
    const max = new Date().getFullYear()
    const min = max - 9
    const year = []

    // eslint-disable-next-line no-plusplus
    for (let i = max; i >= min; i--) {
        year.push(i)
    }

    const listYear = year.map(y => ({
        label: y.toString(),
        value: y,
    }))

    return listYear
}

export const years = arrayOfYears()
