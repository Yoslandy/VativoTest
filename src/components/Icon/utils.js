export const icons = {
    success: "verified_user",
    warning: "warning",
    settings: "settings",
    add_circle_outline: "add_circle_outline",
    eye: "visibility",
    time: "access_time",
    save: "save_outline",
    download: "file_download_outline",
    delete: "delete_outline",
    reset: "key_outline",
    close: "close_outline",
    "eye-slash": "visibility_off",
    error: "do_disturb_alt",
    error_alt: "block_outline",
    login: "login",
    filter_alt: "filter_alt",
    search: "search",
    double_arrow: "keyboard_double_arrow_right",
    arrow: "keyboard_arrow_right",
    queue: "queue",
    show_chart: "show_chart",
    people_alt: "people_alt",
    help_outline: "help_outline",
    account_circle: "account_circle",
    person: "person",
    apartment: "apartment",
    arrow_downward: "arrow_downward",
    arrow_upward: "arrow_upward",
    person_outline: "person_outline",
}

export const styles = {
    outline: "material-icons-outlined",
    default: "material-icons",
}
