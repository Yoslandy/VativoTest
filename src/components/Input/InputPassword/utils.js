export const sizes = {
    small: "h-[40px] w-full",
    big: "h-[43px] w-full",
}
