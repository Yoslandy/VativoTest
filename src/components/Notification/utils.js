export const borders = {
    error: "border-b-semantic-error",
    warning: "border-b-semantic-warning",
    success: "border-b-semantic-success",
}

export const colors = {
    error: "text-semantic-error",
    warning: "text-semantic-warning",
    success: "text-semantic-success",
}
