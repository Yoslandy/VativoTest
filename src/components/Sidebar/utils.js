// TOdo
export const pages = [
    {
        route: "/admin/dashboard",
        iconType: "queue",
        text: "Claims",
    },
    {
        route: "/admin/reports",
        iconType: "show_chart",
        text: "Reports",
    },
    {
        route: "/admin/accounts",
        iconType: "people_alt",
        text: "Accounts",
    },
    {
        route: "/admin/configuration",
        iconType: "settings",
        text: "Configuration",
    },
    {
        route: "/admin/help",
        iconType: "help_outline",
        text: "Help",
    },
]
