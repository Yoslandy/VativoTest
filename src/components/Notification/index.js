/* eslint-disable react/prop-types */
import React, { useCallback } from "react"
import PropTypes from "prop-types"
import { borders, colors } from "./utils"
import Icon from "../Icon"
import SuccessIcon from "../../assets/img/notification-icons/succes.png"

export default function Notification(props) {
    const { appearance, onDismiss, children, title } = props

    const getColorIcon = useCallback(() => colors[appearance], [appearance])

    return (
        <div
            className={`border-b-8 bg-white rounded w-72 p-2 h-full shadow ${borders[appearance]}`}
        >
            <div className="flex w-full">
                <div>
                    {appearance === "success" ? (
                        <img src={SuccessIcon} alt="success" />
                    ) : (
                        <Icon type={appearance} className={getColorIcon()} />
                    )}
                </div>
                <div className="flex flex-col ml-4">
                    <p className="font-medium">{title}</p>
                    <span className="font-normal">{children}</span>
                </div>
                <span onClick={onDismiss} className="absolute right-[27px]">
                    <i className="fas fa-times ml-8 cursor-pointer" />
                </span>
            </div>
        </div>
    )
}

Notification.propTypes = {
    title: PropTypes.string,
    appearance: PropTypes.oneOf(["error", "success", "warning"]),
}

Notification.defaultProps = {
    title: "Title",
    appearance: "success",
}
