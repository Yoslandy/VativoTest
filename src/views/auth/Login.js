import React, { useState, useEffect } from "react"
import { Form, Formik } from "formik"
import { Link, Redirect } from "react-router-dom"
import * as Yup from "yup"
// import { useToasts } from "react-toast-notifications"
import InputPassword from "../../components/Input/InputPassword"
import Input from "../../components/Input"
import Icon from "../../components/Icon"
import logo from "../../assets/img/logo.png"
import { useGraphMutation } from "../../utils/custom-hooks/useGraphClient"
import { LOGIN, LOGOUT } from "../../graphql/mutations/auth"
import { constant } from "../../graphql/constants"

const SignupSchema = Yup.object().shape({
    password: Yup.string().required("Required"),
    email: Yup.string().email("Invalid email").required("Required"),
})

export default function Login() {
    // const { addToast } = useToasts()
    const [isError, setIsError] = useState(false)
    const [isSuccess, setIsSuccess] = useState(false)

    const hideErrorMessage = () => setIsError(false)

    const { callMutation } = useGraphMutation(LOGIN)
    const { callMutation: closeSession } = useGraphMutation(LOGOUT)

    useEffect(() => {
        closeSession()
    }, [])

    const onSubmit = async values => {
        try {
            await callMutation({
                variables: {
                    data: values,
                },
            })
            setTimeout(() => {
                setIsSuccess(true)
            }, 800)
        } catch (error) {
            if (error.message === constant.INVALID_CREDENTIALS) {
                setIsError(true)
            }
        }
    }

    return (
        <div className="container mx-auto px-4 h-full">
            <div className="flex content-center items-center justify-center h-full">
                <div className="px-4 login">
                    <div className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded-xl bg-white border-0">
                        <div className="rounded-t mb-0 px-4 pb-12 pt-6">
                            <img src={logo} alt="logo" />
                        </div>
                        <div className="flex-auto px-4 py-10 pt-0">
                            <Formik
                                initialValues={{ password: "", email: "" }}
                                onSubmit={onSubmit}
                                validationSchema={SignupSchema}
                            >
                                {formik => {
                                    const { errors, values } = formik
                                    const getError = label => {
                                        if (isError) return true
                                        if (!values[label].length) return false
                                        return errors[label]
                                    }

                                    return (
                                        <Form>
                                            <div className="relative w-full mb-6">
                                                <Input
                                                    value={values.email}
                                                    type="email"
                                                    name="email"
                                                    placeholder="Email"
                                                    onFocus={hideErrorMessage}
                                                    error={
                                                        isError || errors.email
                                                    }
                                                />
                                            </div>

                                            <div className="relative w-full mb-6">
                                                <InputPassword
                                                    placeholder="Password"
                                                    name="password"
                                                    value={values.password}
                                                    onFocus={hideErrorMessage}
                                                    error={getError("password")}
                                                />
                                                <p className="input-error-message">
                                                    {isError &&
                                                        "Invalid credentials"}
                                                </p>
                                            </div>
                                            <div />

                                            <div className="text-center mt-6 font-roboto">
                                                <button
                                                    className={`text-[18px] text-white font-medium leading-[21.09px] px-4 py-3 rounded-lg shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 w-full ease-linear transition-all duration-150 justify-between flex items-center ${
                                                        formik.values.email &&
                                                        formik.values.password
                                                            ? "bg-primary-100"
                                                            : "bg-primary-60"
                                                    }`}
                                                    type="submit"
                                                    disabled={
                                                        !formik.values.email &&
                                                        !formik.values.password
                                                    }
                                                >
                                                    <span>Sign in</span>
                                                    <Icon type="login" />
                                                </button>
                                                <Link to="/auth/forgot-password">
                                                    <span className="login-forgot font-medium text-[13px] leading-[15px]">
                                                        {" "}
                                                        Forgot Your Password?
                                                    </span>
                                                </Link>
                                            </div>
                                        </Form>
                                    )
                                }}
                            </Formik>
                        </div>
                    </div>
                </div>
            </div>
            {isSuccess && <Redirect to="/admin" />}
        </div>
    )
}
