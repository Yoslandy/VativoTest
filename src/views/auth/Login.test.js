import React from "react"
import ReactDOM from "react-dom"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import { ToastProvider } from "react-toast-notifications"
import { client } from "../../utils/apollo-client"
import Notification from "../../components/Notification"
import Login from "./Login"
import { ApolloProvider } from "@apollo/client"

describe("Notification", () => {
    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element, node) => {
            return element
        })
    })

    test("Renders Login form correctly", () => {
        const component = renderer.create(
            <MemoryRouter>
                <ApolloProvider client={client}>
                    <ToastProvider components={{ Toast: Notification }}>
                        <Login />
                    </ToastProvider>
                </ApolloProvider>
            </MemoryRouter>
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    afterEach(() => {
        ReactDOM.createPortal.mockClear()
    })
})
