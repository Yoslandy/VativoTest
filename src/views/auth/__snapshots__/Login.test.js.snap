// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Renders Login form correctly 1`] = `
<div
  className="container mx-auto px-4 h-full"
>
  <div
    className="flex content-center items-center justify-center h-full"
  >
    <div
      className="px-4 login"
    >
      <div
        className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded-xl bg-white border-0"
      >
        <div
          className="rounded-t mb-0 px-4 pb-12 pt-6"
        >
          <img
            alt="logo"
            src="logo.png"
          />
        </div>
        <div
          className="flex-auto px-4 py-10 pt-0"
        >
          <form
            action="#"
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="relative w-full mb-6"
            >
              <input
                className="input   border-[1px] border-primary-50"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Email"
                type="email"
                value=""
              />
              <span
                className="formik-error-message"
              />
            </div>
            <div
              className="relative w-full mb-6"
            >
              <div
                className="password-wrapper  border-[1px] border-primary-50 h-[43px] w-full"
              >
                <div
                  className="password-with-asterisk"
                >
                  <input
                    className="password-input h-[41px] text-transparent"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyUp={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <span
                    className="password-with-asterisk-dummy"
                    id="password"
                    style={
                      Object {
                        "display": "inline",
                      }
                    }
                  />
                </div>
                <div
                  className="password-icon"
                >
                  <span
                    className="material-icons "
                    onClick={[Function]}
                    role="button"
                  >
                    visibility
                  </span>
                </div>
              </div>
              <span
                className="formik-error-message"
              />
              <p
                className="input-error-message"
              />
            </div>
            <div />
            <div
              className="text-center mt-6 font-roboto"
            >
              <button
                className="text-[18px] text-white font-medium leading-[21.09px] px-4 py-3 rounded-lg shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 w-full ease-linear transition-all duration-150 justify-between flex items-center bg-primary-60"
                disabled={true}
                type="submit"
              >
                <span>
                  Sign in
                </span>
                <span
                  className="material-icons "
                  onClick={[Function]}
                  role="button"
                >
                  login
                </span>
              </button>
              <a
                href="/auth/forgot-password"
                onClick={[Function]}
              >
                <span
                  className="login-forgot font-medium text-[13px] leading-[15px]"
                >
                   
                  Forgot Your Password?
                </span>
              </a>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
